webroot/js/sentences_lists.set_option.js

```diff
@@ -33,14 +33,13 @@
         $scope.editableChanged = function (oldSetting) {
             $(".is-editable.loader-container").show();
             var listId = $("input[name=editable_by]").attr('data-list-id');
-            var newSetting = $scope.editable;
             var rootUrl = get_tatoeba_root_url();
             $.post(
                 rootUrl + "/sentences_lists/set_option/",
-                { "listId": listId, "option": "editable_by", "value": newSetting },
-                function () {
+                { "listId": listId, "option": "editable_by", "value": $scope.editable },
+                function (response) {
                     $(".is-editable.loader-container").hide();
-                    if (newSetting === "no_one" || oldSetting === "no_one")
+                    if (response.editable_by === "no_one" || oldSetting === "no_one")
                         location.reload();
```

Comment:
Wouldn't it better to avoid the reloading and just change the visibility of the remove icon based on the value of `editable`?

I'm not an AngularJS expert either, but one way to share a model outside of the controller's element subtree is using the `$rootScope` (i.e. a global object). See https://github.com/AndiPersti/tatoeba2/commit/047ce40e89c47847270d150ca5a9e7f76c93bd71

Alternatively we could add a controller to the sentences list and inject a [service](https://docs.angularjs.org/guide/services) in both the `OptionsCtrl` and that new controller.

Reply:
The X button is actually not included in the HTML when the list is not editable at first, so reloading would be necessary if changing from "editable by no one" to "editable by someone".

-------

Assuming we include the HTML every time and just hide it based on the situation, we have the following possibilities:

(1) Since this script is still a hybrid of jQuery and AngularJS, to change the visibility of the X button you could simply do this:
```
if (response.editable_by === "no_one") {
    $(".removeFromList").hide();
} else {
    $(".removeFromList").show();
}
```

(2) If you want to go for the Angular way, using the $rootScope would be a solution (people usually don't recommend using $rootScope but I think in our case it would not be a big deal).

The other way would be to set `ng-controller='optionsCtrl'` on the whole page rather than just the options block. But in that case the name `optionsCtrl` is not very accurate anymore. It should rather be "SentencesListsShowCtrl" and the file could be moved to `/webroot/js/sentences_lists/show.ctrl.js`.

In both cases you'll have to remember to add `ng-if="editable"` on the X button.

-------

That being said, I would suggest to keep the reloading for now. We can later on rewrite this part of the code to be 100% angular (which will allow us to remove the reloading of the page).

Reply:
I, too, think that reloading the page is acceptable for now.